js/src/type.ts

```diff
@@ -333,23 +333,47 @@ export class Decimal extends DataType<Type.Decimal> {
 /** @ignore */
 export type Dates = Type.Date | Type.DateDay | Type.DateMillisecond;
 /** @ignore */
-export interface Date_<T extends Dates = Dates> extends DataType<T> { TArray: Int32Array; TValue: Date; ArrayType: TypedArrayConstructor<Int32Array> }
+type DateType = {
+    [Type.Date]: { TArray: Int32Array | BigInt64Array };
```

Comment:
Hmm... in the Arrow format, there are two date representation: days and milliseconds. So what is this third one?

Reply:
`Type.Date` is the generic one that can be one of the other two. It's the actual physical type (8 in the flatbuffers) while the other two are the actual types people can instantiate.